Eye of the storm

59blargedy opened the pull request:
<!-- place 'x' mark between square [] brackets to affirm: -->
**_I affirm:_**
- [x] that I agree to Project Topaz's [Limited Contributor License Agreement](http://project-topaz.com/blob/release/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

59blargedy committed:
initial set up of eye of the storm bcnm

59blargedy committed:
adding bcnm eye of the storm- mob luas, adjusting thunder and blizzard breath, sql and loot

59blargedy committed:
added comment for later

kaincenteno commented:
the npc_list.sql looks weird, maybe it needed to be rebased? way too many additions/deletions

cocosolos commented on scripts/zones/Chamber_of_Oracles/npcs/Armoury_Crate.lua:
```diff
@@ -67,6 +67,76 @@ local loot =
+        },
+          
+        {    
+            {itemid = 65535, droprate = 1000, amount = 4000},  -- 4k Gil, really should be 24k/total number of players. 
```
Amount should just be 24k, loot handling will split it automatically.
https://youtu.be/TlvThE0sM14?t=950

59blargedy replied:
got it, wasn't sure how that split worked and thought i would need some getplayercount shenanigans i wasnt prepared yet to do. much easier, thank you!

cocosolos commented on sql/bcnm_battlefield.sql:
```diff
@@ -630,6 +630,22 @@ INSERT INTO `bcnm_battlefield` VALUES (196,2,17465362,3);
+INSERT INTO `bcnm_battlefield` VALUES (198,3,17465403,3);
+INSERT INTO `bcnm_battlefield` VALUES (198,3,17465404,3);
+INSERT INTO `bcnm_battlefield` VALUES (198,3,17465405,3);
+INSERT INTO `bcnm_battlefield` VALUES (198,3,17465406,3);
```
Duplicate entries?

59blargedy replied:
yep :( thanks for catching

59blargedy committed:
fixing errors from comments provided (thanks!)

59blargedy commented:
> 
> 
> the npc_list.sql looks weird, maybe it needed to be rebased? way too many additions/deletions

I'm not sure what i did or how to fix it. Do you know?

59blargedy committed:
copied down original raw sql file and made changes

59blargedy commented:
@kaincenteno i figured out what i did wrong and how to fix it, sorry about that :(

59blargedy committed:
battlefield status and typo correction

KnowOne134 commented on scripts/zones/Chamber_of_Oracles/mobs/Radiant_Wyvern.lua:
```diff
@@ -0,0 +1,22 @@
+-- KSNM: Eye of the Storm
+-----------------------------------
+require("scripts/globals/status")
+require("scripts/globals/monstertpmoves")
```
monstertpmoves isnt needed as a require in all the mob scripts

59blargedy replied:
they dont have a special ability before death, wiki is just referring to their unique breath attacks. will look at other alterations - wyverns already have boosted movement speed of 50 not sure if they need more but will compare against vid.

KnowOne134 reviewed (commented):
wiki says they have significant Movement Speed, heavy sleep resistance, and a special ability right before each die

59blargedy committed:
adjusted wyvern speed - checked other mobs in family off of npclog files and they are all 75 movement
removed monstertpmoveinclude
i am already unable to sleep without ele seal so it seems to work as intended without me doing anything. i am admittedly mystified. lullaby makes more sense bc of light resist.

KnowOne134 reviewed (commented):
we sure all the mobs in that family are movement of 75 before i say merge? i went to my database and filtered all pools that had family 266 and pulled up 30 pools, such as Vouivre, Pyrodrake, Hurricane_Wyvern, Bune. to name a few

59blargedy commented:
I checked: riverne, ifrit, gustav, abyssea-grau nms and standard mobs are
75 per npc database. If we need all 30 pools I can go validate each one if
preferred.



On Fri, Mar 27, 2020 at 12:14 PM KnowOne134 <notifications@github.com>
wrote:

> *@KnowOne134* commented on this pull request.
>
> we sure all the mobs in that family are movement of 75 before i say merge?
> i went to my database and filtered all pools that had family 266 and pulled
> up 30 pools, such as Vouivre, Pyrodrake, Hurricane_Wyvern, Bune. to name a
> few
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/project-topaz/topaz/pull/437#pullrequestreview-383003393>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMRSUMGBEOI47MYV5BIFMMDRJTGH5ANCNFSM4LSLR45A>
> .
>

KnowOne134 reviewed (approved):
Awaiting 2nd approval

ibm2431 commented on sql/mob_skill_lists.sql:
```diff
@@ -3461,3 +3461,24 @@ INSERT INTO `mob_skill_lists` VALUES ('Siren',1010,3512); -- Sonic Buffet
+INSERT INTO `mob_skill_lists` VALUES ('Thunder_Wyvern',1011,816); -- Fang RUsh
+INSERT INTO `mob_skill_lists` VALUES ('Thunder_Wyvern',1011,817); -- Dread Shriek
+INSERT INTO `mob_skill_lists` VALUES ('Thunder_Wyvern',1011,818); -- Tail Crush
+INSERT INTO `mob_skill_lists` VALUES ('Thunder_Wyvern',1011,820); -- Thunder Breath
```
Make sure the name matches the mob name (`Lightning_Wyvern`), and sort the skill list ID (1011) to be above the Chaos Wyvern (1012) and I'll hit the button.

ibm2431 commented on scripts/zones/Chamber_of_Oracles/mobs/Blizzard_Wyvern.lua:
```diff
@@ -0,0 +1,21 @@
+end
+
+function onMobFight(mob, target)
+    mob:setMod(tpz.mod.REGAIN, 100)
```
Set mod in `onMobFight` or `onMobEngaged`?

59blargedy replied:
I'm not sure which is right - they basically use it as soon as they see you. if onmobfight is better, will do there. it works as is in terms of testing, but rather do it in the right place if they are different.

ibm2431 replied:
onMobEngaged is when the mob aggros (and also when it reaggros); onMobFight is constantly - I believe it's every attack round

59blargedy replied:
oh yes, i remember now. most definitely how i have it, when it aggros in onmobengaged

59blargedy committed:
updated mob_skill_lists to reflect what i actually put in the database >_>

59blargedy committed:
moving regain check to onmobengaged
